### PR TITLE
feat(Badge): Add color support

### PR DESCRIPTION
**Pull request by gjou-tlnd**
**What is the problem this PR is trying to solve?**

Colored badge are needed for faceted search fillters
https://www.sketch.com/s/bcd371d3-ebbc-40dd-a3cc-2db395fdb216/a/dJbdPn

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Badge/Badge.scss [hunk 1]
```diff
@@ -257,4 +277,24 @@ $tc-badge-disabled-opacity: 0.62;
 			opacity: $tc-badge-disabled-opacity;
 		}
 	}
+
+	&-valid-filter {
+		@include colored-badge(adjust_color($rio-grande, $alpha: -0.7), darken($rio-grande, 30%));
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:282]: you could use `shade()` and `tint()`

File: packages/components/src/Badge/Badge.component.js [hunk 2]
```diff
@@ -13,6 +13,14 @@ const SIZES = {
 	small: 'small',
 };
 
+const COLORS = {
+	VALID_FILTER: 'valid-filter',
+	INVALID_FILTER: 'invalid-filter',
+	EMPTY_FILTER: 'empty-filter',
+	PATTERN_FILTER: 'pattern-filter',
+	VALUE_FILTER: 'value-filter',
+};
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.component.js:22]:
```suggestion
const TYPES = {
	VALID: 'valid',
	INVALID: 'invalid',
	EMPTY: 'empty',
	PATTERN: 'patterns',
	VALUE: 'value',
};
```

File: packages/components/src/Badge/Badge.component.js [hunk 3]
```diff
@@ -71,6 +79,7 @@ function Badge({
 	selected = false,
 	style,
 	white,
+	color,
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.component.js:82]:
```suggestion
	type,
```

File: packages/components/src/Badge/Badge.component.js [hunk 4]
```diff
@@ -81,6 +90,7 @@ function Badge({
 		'tc-badge-readonly': !onDelete,
 		'tc-badge-aslink': aslink,
 		'tc-badge-edit': onDelete && onSelect,
+		[`tc-badge-${color}`]: !!color,
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.component.js:93]:
```suggestion
		[`tc-badge--${type}`]: !!type,
```

File: packages/components/src/Badge/Badge.component.js [hunk 5]
```diff
@@ -127,7 +137,9 @@ Badge.propTypes = {
 	selected: PropTypes.bool,
 	style: PropTypes.object,
 	white: PropTypes.bool,
+	color: PropTypes.oneOf(Object.values(COLORS)),
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.component.js:140]:
```suggestion
	color: PropTypes.oneOf(Object.values(TYPES)),
```

File: packages/components/src/Badge/Badge.component.js [hunk 6]
```diff
@@ -127,7 +137,9 @@ Badge.propTypes = {
...
 };
 
 Badge.SIZES = SIZES;
+Badge.COLORS = COLORS;
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.component.js:144]:
```suggestion
Badge.TYPES = TYPES;
```

File: packages/components/src/Badge/Badge.scss [hunk 7]
```diff
@@ -257,4 +277,24 @@ $tc-badge-disabled-opacity: 0.62;
 			opacity: $tc-badge-disabled-opacity;
 		}
 	}
+
+	&-valid-filter {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:281]:
```suggestion
	&--valid{
```

File: packages/components/src/Badge/Badge.scss [hunk 8]
```diff
@@ -257,4 +277,24 @@ $tc-badge-disabled-opacity: 0.62;
...
+		@include colored-badge(adjust_color($rio-grande, $alpha: -0.7), darken($rio-grande, 30%));
+	}
+
+	&-invalid-filter {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:285]:
```suggestion
	&--invalid{
```

File: packages/components/src/Badge/Badge.scss [hunk 9]
```diff
@@ -257,4 +277,24 @@ $tc-badge-disabled-opacity: 0.62;
...
+		@include colored-badge(adjust_color($chestnut-rose, $alpha: -0.7), darken($chestnut-rose, 30%));
+	}
+
+	&-empty-filter {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:289]:
```suggestion
	&--empty {
```

File: packages/components/src/Badge/Badge.scss [hunk 10]
```diff
@@ -257,4 +277,24 @@ $tc-badge-disabled-opacity: 0.62;
...
+		@include colored-badge($alto, $dove-gray);
+	}
+
+	&-pattern-filter {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:293]:
```suggestion
	&--pattern {
```

File: packages/components/src/Badge/Badge.scss [hunk 11]
```diff
@@ -257,4 +277,24 @@ $tc-badge-disabled-opacity: 0.62;
...
+		@include colored-badge(adjust_color($jaffa, $alpha: -0.7), darken($jaffa, 30%));
+	}
+
+	&-value-filter {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:297]:
```suggestion
	&--value{
```

File: packages/components/src/Badge/Badge.stories.js [hunk 12]
```diff
@@ -39,687 +39,740 @@ function onSelect(name) {
...
-					/>
-				</div>
-				<div style={columnStyle}>
-					<Badge
-						label={longStr}
-						display={Badge.SIZES.small}
-						category="Cat"
-						white
-						{...onSelect('B')}
+						color={value}
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.stories.js:770]:
```suggestion
						type={value}
```

File: packages/faceted-search/src/components/BasicSearch/BasicSearch.component.js [hunk 13]
```diff
@@ -37,15 +37,16 @@ const BasicSearch = ({
 	customBadgesDictionary,
 	customOperatorsDictionary,
 	initialFilterValue,
+	useColors = false,
```
Comment (vbocquet) [inline packages/faceted-search/src/components/BasicSearch/BasicSearch.component.js:40]: Why have a conditional feature boolean here? I think we need to provide all badges supports by default.
It's better to modify the badgesDefinitions instead, to be sure that FacetedSearch is as dumb as possible.

**Review by vbocquet — COMMENTED** [review]
I make a first read of the PR but i'm a little lost of this PR, it's not a new badge right? Can you link a public link to sketch? The one on the PR description is private

File: packages/faceted-search/src/components/BasicSearch/BasicSearch.component.js (hunk 13 above)
Comment (gjou-tlnd) [reply packages/faceted-search/src/components/BasicSearch/BasicSearch.component.js]: > I make a first read of the PR but i'm a little lost of this PR, it's not a new badge right? Can you link a public link to sketch? The one on the PR description is private

the point is to add colors to existing badges and new badges for TDS and TDP
image

I split the PR, this one is now about Badge component only, faceted search code will be in another one

**Review by jsomsanith-tlnd — APPROVED** [review]
I approve, implementation is ok, but we have contrast ratio issues on the guildeline.
Can you check with our visual designer please ?

**Review by tlnd-mhuchet — APPROVED** [review]

File: packages/components/src/Badge/Badge.scss [hunk 14]
```diff
@@ -20,6 +20,27 @@ $tc-badge-small-padding: 0.8rem;
...
+@mixin colored-badge($background-color, $color) {
+	.tc-badge-button {
+		background-color: $background-color;
+		color: $color;
+
+		.tc-badge-separator {
+			background-color: $color;
+		}
+
+		.tc-badge-category,
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss:32]: Could you extract this style reset and use it once with `[class^="tc-badge--"]` or something like that:
```js
const containerClasses = theme('tc-badge', displayClass, className, {
  //...
  'tc-badge--colored': !!type,
  [`tc-badge--${type}`]: !!type,
});```

**Review by frassinier — APPROVED** [review]

**Comment by gjou-tlnd** [comment]
> I approve, implementation is ok, but we have contrast ratio issues on the guildeline.
> Can you check with our visual designer please ?

specs are different between frontify & sketch.. fixed it

**Pull request merged**